File: mobile/test/item_display_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:mobile/core/api_client.dart';

void main() {
  InventoryItem item({String? partNumber}) => InventoryItem(
    itemId: 'item-1',
    name: 'Washer for 14mm Bearing with 8mm REX shaft',
    category: 'Supplies',
    quantity: 1,
    location: 'Fastener db',
    createdAt: DateTime(2026, 9, 7),
    partNumber: partNumber,
  );

  test('part number is the primary display label when present', () {
    final inventoryItem = item(partNumber: 'PN-F280');

    expect(inventoryItem.displayName, 'PN-F280');
    expect(
      inventoryItem.displayDescription,
      'Washer for 14mm Bearing with 8mm REX shaft',
    );
  });

  test('name remains primary when no part number exists', () {
    final inventoryItem = item();

    expect(
      inventoryItem.displayName,
      'Washer for 14mm Bearing with 8mm REX shaft',
    );
    expect(inventoryItem.displayDescription, isNull);
  });
}

File: frontend/src/__tests__/item-display.test.ts
import { itemDisplayDescription, itemDisplayName } from "@/lib/api";

describe("inventory item display hierarchy", () => {
  it("uses the part number as the primary label", () => {
    const item = {
      name: "Washer for 14mm Bearing with 8mm REX shaft",
      part_number: "PN-F280",
    };

    expect(itemDisplayName(item)).toBe("PN-F280");
    expect(itemDisplayDescription(item)).toBe(
      "Washer for 14mm Bearing with 8mm REX shaft",
    );
  });

  it("falls back to the item name when no part number exists", () => {
    const item = {
      name: "Unnumbered washer",
      part_number: null,
    };

    expect(itemDisplayName(item)).toBe("Unnumbered washer");
    expect(itemDisplayDescription(item)).toBeNull();
  });
});
